tests/bluetooth/bsim_bt/bsim_test_audio/src/unicast_client_test.c

```diff
@@ -0,0 +1,484 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(CONFIG_BT_BAP) && defined(CONFIG_BT_AUDIO_UNICAST_CLIENT)
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/audio.h>
+#include "common.h"
+
+extern enum bst_result_t bst_result;
+
+static struct bt_audio_chan g_channels[CONFIG_BT_BAP_ASE_SNK_COUNT];
+static struct bt_audio_capability *g_remote_capabilities[CONFIG_BT_BAP_PAC_COUNT];
+static struct bt_audio_ep *g_sinks[CONFIG_BT_BAP_ASE_SNK_COUNT];
+static struct bt_conn *g_conn;
+
+#define LC3_PRESET(_name, _codec, _qos) \
+	{ \
+		.name = _name, \
+		.codec = _codec, \
+		.qos = _qos, \
+	}
+
+struct lc3_preset {
+	const char *name;
+	struct bt_codec codec;
+	struct bt_codec_qos qos;
+};
+
+/* Mandatory support preset by both client and server */
+static struct lc3_preset preset_16_2_1 =
+	LC3_PRESET("16_2_1",
+		   BT_CODEC_LC3_CONFIG_16_2,
+		   BT_CODEC_LC3_QOS_10_INOUT_UNFRAMED(40u, 2u, 10u, 40000u));
+
+CREATE_FLAG(flag_connected);
+CREATE_FLAG(flag_mtu_exchanged);
+CREATE_FLAG(flag_sink_discovered);
+CREATE_FLAG(flag_chan_configured);
+CREATE_FLAG(flag_chan_qos);
+CREATE_FLAG(flag_chan_enabled);
+
+static void print_codec(const struct bt_codec *codec)
+{
+	printk("codec 0x%02x cid 0x%04x vid 0x%04x count %u\n",
+	       codec->id, codec->cid, codec->vid, codec->data_count);
+
+	for (uint8_t i = 0; i < codec->data_count; i++) {
+		printk("data #%zu: type 0x%02x len %u\n",
+		       i, codec->data[i].data.type,
+		       codec->data[i].data.data_len);
+		print_hex(codec->data[i].data.data,
+			  codec->data[i].data.data_len -
+				sizeof(codec->data[i].data.type));
+		printk("\n");
+	}
+
+	for (uint8_t i = 0; i < codec->meta_count; i++) {
+		printk("meta #%zu: type 0x%02x len %u\n",
+		       i, codec->meta[i].data.type,
+		       codec->meta[i].data.data_len);
+		print_hex(codec->meta[i].data.data,
+			  codec->meta[i].data.data_len -
+				sizeof(codec->meta[i].data.type));
+		printk("\n");
+	}
+}
+
+static void print_qos(struct bt_codec_qos *qos)
+{
+	printk("QoS: dir 0x%02x interval %u framing 0x%02x phy 0x%02x sdu %u "
+	       "rtn %u latency %u pd %u\n",
+	       qos->dir, qos->interval, qos->framing, qos->phy, qos->sdu,
+	       qos->rtn, qos->latency, qos->pd);
+}
+
+static struct bt_audio_chan *lc3_config(struct bt_conn *conn,
+					struct bt_audio_ep *ep,
+					struct bt_audio_capability *cap,
+					struct bt_codec *codec)
+{
+	printk("ASE Codec Config: conn %p ep %p cap %p\n", conn, ep, cap);
+
+	print_codec(codec);
+
+	for (size_t i = 0; i < ARRAY_SIZE(g_channels); i++) {
+		if (g_channels[i].conn == NULL) {
+			return &g_channels[i];
+		}
+	}
+
+	FAIL("No channels available\n");
+
+	return NULL;
+}
+
+static int lc3_reconfig(struct bt_audio_chan *chan,
+			struct bt_audio_capability *cap,
+			struct bt_codec *codec)
+{
+	printk("ASE Codec Reconfig: chan %p cap %p\n", chan, cap);
+
+	print_codec(codec);
+
+	SET_FLAG(flag_chan_configured);
```

Comment:
`lc3` here refers to the capability that was registered that uses these callback functions (i.e. a capability that uses the LC3 codec, hence the naming). 

The naming was taken from the BAP shell module, but it may be better to rename the callback functions to something else.